packages/@tinacms/toolkit/src/packages/fields/plugins/MdxFieldPlugin/updated/plate/plugins/create-slash-plugin/index.tsx

```diff
@@ -0,0 +1,180 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import React from 'react'
+import {
+  createPluginFactory,
+  insertNodes,
+  getPlugin,
+  unwrapNodes,
+  findNode,
+  getPluginType,
+  PlateEditor,
+} from '@udecode/plate-core'
+import { computePosition, flip, shift } from '@floating-ui/dom'
+import { Element, Transforms } from 'slate'
+import { useFocused, useSelected } from 'slate-react'
+import { SearchAutocomplete } from '../ui/combobox'
+import { insertMDX } from '../create-mdx-plugins'
+import type { MdxTemplate } from '../../types'
+import { helpers } from '../core/common'
+
+export const ELEMENT_MAYBE_MDX = 'maybe_mdx'
+
+export const isSelectionInMaybeMDX = (editor: PlateEditor) =>
+  findMaybeMDX(editor) !== undefined
+
+export const findMaybeMDX = (editor: PlateEditor) =>
+  findNode(editor, {
+    match: { type: getPluginType(editor, ELEMENT_MAYBE_MDX) },
+  })
+
+/**
+ * `ELEMENT_MAYBE_MDX` is inserted by a forward slash (`/`)
+ * that indicates we may drop in an MDX component from the combobox.
+ * This can probably be customized to whatever command
+ * the developer wants (eg. `<`)
+ */
+export const withMaybeMDX = (editor) => {
+  const { type } = getPlugin(editor, ELEMENT_MAYBE_MDX)
+  const { insertText } = editor
+  editor.insertText = (text) => {
+    if (isSelectionInMaybeMDX(editor)) {
+      return Transforms.insertText(editor, text)
+    }
+    if (!editor.selection || text !== '/') {
+      return insertText(text)
+    }
+    if (helpers.currentNodeSupportsMDX(editor)) {
+      insertNodes(editor, {
+        type,
+        children: [{ type: 'text', text: '/' }],
+      })
+    } else {
+      return insertText(text)
+    }
+  }
+
+  return editor
+}
+
+export const createSlashPlugin = createPluginFactory({
+  key: ELEMENT_MAYBE_MDX,
+  isElement: true,
+  isInline: true,
+  withOverrides: withMaybeMDX,
+  component: (props) => <SlashCombobox {...props} />,
+})
+
+const SlashCombobox = (props: {
+  editor: PlateEditor
+  element: Element
+  attributes: object
+  className: string
+  children: React.ReactNode
+  templates: MdxTemplate[]
+}) => {
+  const baseRef = React.useRef<HTMLSpanElement | null>()
+  const ref = React.useRef<HTMLSpanElement | null>()
+  const focused = useFocused()
+  const selected = useSelected()
+  const onCancel = () => {
+    unwrapNodes(props.editor, {
+      match: (n) => Element.isElement(n) && n.type === ELEMENT_MAYBE_MDX,
+    })
+  }
+
+  const onValue = (value: MdxTemplate) => {
+    // Effectively replaces the existing node with the MDX element
+    Transforms.removeNodes(props.editor, {
+      match: (n) => {
+        // @ts-ignore bad type from slate
+        if (Element.isElement(n) && n.type === ELEMENT_MAYBE_MDX) {
+          return true
+        }
+      },
+    })
+    insertMDX(props.editor, value)
+  }
```

Comment:
Once we've selected a JSX embed, [unwrap](https://docs.slatejs.org/api/transforms#transforms.unwrapnodes-editor-editor-options) the `ELEMENT_MAYBE_MDX` node. When we first type `/`, the element is turned into an `ELEMENT_MAYBE_MDX` node (that happens [here](https://github.com/tinacms/tinacms/pull/2473/files#diff-e72a6523ea2b2a181086ea7b625bfdc84d0bdc2e69c70ebbe48e9a23bee101d2R64-R66)):
<img width="308" alt="Screen Shot 2022-01-27 at 3 18 34 PM" src="https://user-images.githubusercontent.com/5414297/151459388-9024320e-07a7-4ca7-81a0-b8ea1e6ef5a6.png">

```
[
  {
    "type": "p",
    "children": [
      {
        "text": "Hello, "
      },
      {
        "type": "maybe_mdx",
        "children": [
          {
            "type": "text",
            "text": "/Da",
          }
        ],
      },
      {
        "text": ""
      }
    ],
  }
]
```
And then the `insertMDX` action kicks in, inserting the 'Date & Time' element:

<img width="243" alt="Screen Shot 2022-01-27 at 3 19 29 PM" src="https://user-images.githubusercontent.com/5414297/151459514-072cce89-3a08-4dd7-adb4-90b49732f833.png">

```
[
  {
    "type": "p",
    "children": [
      {
        "text": "Hello, "
      },
      {
        "type": "mdxJsxTextElement",
        "name": "DateTime",
        "children": [
          {
            "text": ""
          }
        ],
        "props": {},
      },
      {
        "text": ""
      }
    ],
  }
]
```